docs/source/tutorials/basics/basics_setup.rst

```diff
@@ -37,157 +37,13 @@ then computing the error on the validation set. This is what we will do here,
-       train_sampler = SubsetRandomSampler(train_idx)
-       valid_sampler = SubsetRandomSampler(valid_idx)
-       batch_size = config["batch_size"]
-       train_loader = torch.utils.data.DataLoader(
-           data_train, batch_size=batch_size, sampler=train_sampler, drop_last=True
-       )
-       valid_loader = torch.utils.data.DataLoader(
-           data_train, batch_size=batch_size, sampler=valid_sampler, drop_last=True
-       )
-       return train_loader, valid_loader
-
-
-   def model_and_optimizer(config):  # [4]
-       n_units_1 = config["n_units_1"]
-       n_units_2 = config["n_units_2"]
-       dropout_1 = config["dropout_1"]
-       dropout_2 = config["dropout_2"]
-       learning_rate = config["learning_rate"]
-       weight_decay = config["weight_decay"]
-       # Define the network architecture
-       comp_list = [
-           nn.Linear(28 * 28, n_units_1),
-           nn.Dropout(p=dropout_1),
-           nn.ReLU(),
-           nn.Linear(n_units_1, n_units_2),
-           nn.Dropout(p=dropout_2),
-           nn.ReLU(),
-           nn.Linear(n_units_2, 10),
-       ]
-       model = nn.Sequential(*comp_list)
-       optimizer = torch.optim.Adam(
-           model.parameters(), lr=learning_rate, weight_decay=weight_decay
-       )
-       criterion = nn.CrossEntropyLoss()
-       return {"model": model, "optimizer": optimizer, "criterion": criterion}
-
-
-   def train_model(config, state, train_loader):
-       model = state["model"]
-       optimizer = state["optimizer"]
-       criterion = state["criterion"]
-       batch_size = config["batch_size"]
-       model.train()
-       for data, target in train_loader:
-           optimizer.zero_grad()
-           output = model(data.view(batch_size, -1))
-           loss = criterion(output, target)
-           loss.backward()
-           optimizer.step()
-
-
-   def validate_model(config, state, valid_loader):
-       batch_size = config["batch_size"]
-       model = state["model"]
-       model.eval()
-       correct = 0
-       total = 0
-       for data, target in valid_loader:
-           output = model(data.view(batch_size, -1))
-           _, predicted = torch.max(output.data, 1)
-           total += target.size(0)
-           correct += (predicted == target).sum().item()
-       return correct / total  # Validation accuracy
-
-
-   def objective(config):  # [1]
-       # Download data
-       data_train = download_data(config)
-       # Report results to Syne Tune
-       report = Reporter()
-       # Split into training and validation set
-       train_loader, valid_loader = split_data(config, data_train)
-       # Create model and optimizer
-       state = model_and_optimizer(config)
-       # Training loop
-       for epoch in range(1, config["epochs"] + 1):
-           train_model(config, state, train_loader)
+.. literalinclude:: code/traincode_report_end.py
+   :caption: traincode_report_end.py
+   :lines: 13-
```

Comment:
This whole PR is super cool! This will really make it easier to keep everything up-to-date